### PR TITLE
Wildcard pattern match in Amazon EventBridge rule (S3-EventBridge-SQS)

### PR DESCRIPTION
**Pull request by biswanathmukherjee**
*Issue #, if available:*

*Description of changes:*
The SAM template deploys an Amazon S3 bucket, an Amazon EventBridge rule with wildcard pattern match and an Amazon EventBridge target as Amazon SQS Queue. Contains sample scripts for test.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: s3-eventbridge-wildcard-sam/upload-to-s3.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+if [ -f "$CSV_FILE" ]; then
+  aws s3 cp $CSV_FILE s3://$BUCKET/$FOLDER/
+else
+  echo "File $CSV_FILE does not exist"
+  exit 1
+fi
+
+
+# S3 folder for ABC
```
Comment (bfreiberg) [inline s3-eventbridge-wildcard-sam/upload-to-s3.sh:27]:
```suggestion
# S3 folder for XYZ 
```

File: s3-eventbridge-wildcard-sam/upload-to-s3.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+else
+  echo "File $CSV_FILE does not exist"
+  exit 1
+fi
+
+
+# S3 folder for ABC 
+FOLDER="$DATE/$TIMESTAMP/XYZ"
+
+# Create folder structure for ABC data in S3
```
Comment (bfreiberg) [inline s3-eventbridge-wildcard-sam/upload-to-s3.sh:30]:
```suggestion
# Create folder structure for XYZ data in S3
```

File: s3-eventbridge-wildcard-sam/upload-to-s3.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+aws s3api put-object --bucket $BUCKET --key $FOLDER/
+
+# Upload ABC data to S3
```
Comment (bfreiberg) [inline s3-eventbridge-wildcard-sam/upload-to-s3.sh:33]:
```suggestion
# Upload XYZ data to S3 
```

File: s3-eventbridge-wildcard-sam/example-pattern.json [hunk 4]
```diff
@@ -0,0 +1,61 @@
+{
+  "title": "Wildcard pattern match in Amazon EventBridge rule",
```
Comment (bfreiberg) [inline s3-eventbridge-wildcard-sam/example-pattern.json:2]:
```suggestion
  "title": "S3 to SQS with EventBridge with wildcard pattern match rule",
```

File: s3-eventbridge-wildcard-sam/example-pattern.json [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  "description": "The SAM template deploys an Amazon S3 bucket, an Amazon EventBridge rule with wildcard pattern match and an Amazon EventBridge target as Amazon SQS Queue.",
```
Comment (bfreiberg) [inline s3-eventbridge-wildcard-sam/example-pattern.json:3]: Please shorten it to 150 characters or less

File: s3-eventbridge-wildcard-sam/example-pattern.json [hunk 6]
```diff
@@ -0,0 +1,61 @@
+{
+  "title": "Wildcard pattern match in Amazon EventBridge rule",
+  "description": "The SAM template deploys an Amazon S3 bucket, an Amazon EventBridge rule with wildcard pattern match and an Amazon EventBridge target as Amazon SQS Queue.", 
+  "language": "YAML",
+  "level": "200",
+  "framework": "SAM",
+  "introBox": {
+    "headline": "How it works",
+    "text": [
+      "This template creates an S3 bucket that publishes events to Amazon EventBridge, an EventBridge rule with wildcard filter to select only the desired events from dynamic subfolders in Amazon S3 bucket and also an Amazon SQS Queue as the target.",
```
Comment (bfreiberg) [inline s3-eventbridge-wildcard-sam/example-pattern.json:10]:
```suggestion
      "This template creates an S3 bucket that publishes events to Amazon EventBridge, an EventBridge rule with wildcard filter to select only the desired events from dynamic subfolders in Amazon S3 bucket and also an Amazon SQS queue as the target.",
```

File: s3-eventbridge-wildcard-sam/example-pattern.json [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  "gitHub": {
+    "template": {
+      "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/s3-eventbridge-wildcard-sam",
+      "templateURL": "serverless-patterns/s3-eventbridge-wildcard-sam",
+      "projectFolder": "s3-eventbridge-wildcard-sam",
+      "templateFile": "template.yaml"
+    }
+  },
+  "resources": {
+    "bullets": [
```
Comment (bfreiberg) [inline s3-eventbridge-wildcard-sam/example-pattern.json:25]: Please include a link to S3 event notifications

File: s3-eventbridge-wildcard-sam/README.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+         "ETag": "\"d41d8cd98f00b204e9800998ecf8427e\"",
+         "ServerSideEncryption": "AES256"
+      }
+      upload: ./abc-data.csv to s3://{your-bucket-name}/2024-01-17/1705479398/ABC/abc-data.csv
+      {
+         "ETag": "\"d41d8cd98f00b204e9800998ecf8427e\"",
+         "ServerSideEncryption": "AES256"
+      }
+      upload: ./xyz-data.csv to s3://{your-bucket-name}/2024-01-17/1705479398/XYZ/xyz-data.csv
+      ABC and XYZ data uploaded to S3 bucket serverlessland-s3-eb-sqs-wildcard
```
Comment (bfreiberg) [inline s3-eventbridge-wildcard-sam/README.md:67]: please remove the bucket name

File: s3-eventbridge-wildcard-sam/README.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
+# S3 to EventBridge to SQS with wildcard pattern match
+
+Publish events directly from Amazon S3 to EventBridge. There are multiple dynamic subfolders within the S3 buckets. EventBridge rule uses wildcard filters to select the required events from all the events and sends message to an Amazon SQS Queue.
```
Comment (bfreiberg) [inline s3-eventbridge-wildcard-sam/README.md:3]: Please reword it in a way that explains what the pattern creates and what it does.

File: s3-eventbridge-wildcard-sam/upload-to-s3.sh (hunk 1 above)
Comment (biswanathmukherjee) [reply s3-eventbridge-wildcard-sam/upload-to-s3.sh]: Fixed.

File: s3-eventbridge-wildcard-sam/upload-to-s3.sh (hunk 2 above)
Comment (biswanathmukherjee) [reply s3-eventbridge-wildcard-sam/upload-to-s3.sh]: Fixed.

File: s3-eventbridge-wildcard-sam/upload-to-s3.sh (hunk 3 above)
Comment (biswanathmukherjee) [reply s3-eventbridge-wildcard-sam/upload-to-s3.sh]: Fixed.

File: s3-eventbridge-wildcard-sam/example-pattern.json (hunk 4 above)
Comment (biswanathmukherjee) [reply s3-eventbridge-wildcard-sam/example-pattern.json]: Fixed.

File: s3-eventbridge-wildcard-sam/example-pattern.json (hunk 5 above)
Comment (biswanathmukherjee) [reply s3-eventbridge-wildcard-sam/example-pattern.json]: Fixed.

File: s3-eventbridge-wildcard-sam/example-pattern.json (hunk 6 above)
Comment (biswanathmukherjee) [reply s3-eventbridge-wildcard-sam/example-pattern.json]: Fixed.

File: s3-eventbridge-wildcard-sam/example-pattern.json (hunk 7 above)
Comment (biswanathmukherjee) [reply s3-eventbridge-wildcard-sam/example-pattern.json:25]: Added.

File: s3-eventbridge-wildcard-sam/README.md (hunk 8 above)
Comment (biswanathmukherjee) [reply s3-eventbridge-wildcard-sam/README.md]: Fixed.

File: s3-eventbridge-wildcard-sam/README.md (hunk 9 above)
Comment (biswanathmukherjee) [reply s3-eventbridge-wildcard-sam/README.md]: Updated.

**Comment by biswanathmukherjee** [comment]
Thanks @bfreiberg for the detailed review. I have updated the PR as per your review comment. Can you please review it?

**Comment by bfreiberg** [comment]
Looks good thanks for your contribution.

**Comment by biswanathmukherjee** [comment]
Thanks @bfreiberg. It seems that the PR is not merged yet with the main branch. Can you please help?

**Comment by bfreiberg** [comment]
The pattern will be merged soon by one of the developer advocates. This process can take a few days but is usually quicker.

**Pull request merged**